ratelimit.go

```diff
@@ -89,18 +105,16 @@ func New(rate int, opts ...Option) Limiter {
 
 // WithClock returns an option for ratelimit.New that provides an alternate
 // Clock implementation, typically a mock Clock for testing.
-func WithClock(clock Clock) Option {
-	return func(l *limiter) {
-		l.clock = clock
+func WithClock(clock Clock) option {
+	return func(c *config) {
+		c.clock = clock
 	}
 }
 
 // WithoutSlack is an option for ratelimit.New that initializes the limiter
 // without any initial tolerance for bursts of traffic.
-var WithoutSlack Option = withoutSlackOption
-
-func withoutSlackOption(l *limiter) {
-	l.maxSlack = 0
+var WithoutSlack option = func(c *config) {
```

Comment:
why not just provide `WithSlack(slack time.Duration)`? You can easily pass in `WithSlack(0)` while also allowing the caller to set it to any other value.

Reply:
I do agree this would make more sense.

The reasoning was:
- I tried to keep changes here minimal - keeping the same API
- the WithSlack(5) actually wouldn't work, we just don't respect it.

So I thought I'd fix the slack with actual targeted changes.

Reply:
i see. agreed that we should keep the same api. we can add new options in a different diff then